code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageFragment.java

```diff
@@ -172,7 +174,55 @@ public void onCreate(final Bundle savedInstanceState) {
                                 .getApplicationContext(),
                         webViewGestureListener);
 
-        setStyle(DialogFragment.STYLE_NORMAL, android.R.style.Theme_Translucent_NoTitleBar);
+        setStyle(DialogFragment.STYLE_NO_FRAME, android.R.style.Theme_Translucent_NoTitleBar);
+    }
+
+    @Override
+    public Dialog onCreateDialog(final Bundle savedInstanceState) {
+        final Activity parentActivity = getActivity();
+        final int fragmentTheme = getTheme();
+        final Dialog defaultDialog = super.onCreateDialog(savedInstanceState);
+
+        if (parentActivity == null || message == null) {
+            Log.trace(
+                    ServiceConstants.LOG_TAG,
+                    TAG,
+                    "%s (%s), returning a default Dialog object.",
+                    parentActivity == null ? "Parent Activity" : "Message",
+                    UNEXPECTED_NULL_VALUE);
+            return defaultDialog;
+        }
+
+        final MessageSettings messageSettings = message.getMessageSettings();
+        if (messageSettings == null) {
+            Log.trace(
+                    ServiceConstants.LOG_TAG,
+                    TAG,
+                    "%s (MessageSettings), returning a default Dialog object.",
+                    UNEXPECTED_NULL_VALUE);
+            return defaultDialog;
+        }
+
+        final boolean uiTakeoverEnabled = messageSettings.getUITakeover();
+        return new Dialog(parentActivity, fragmentTheme) {
+            @Override
+            public boolean onTouchEvent(@NonNull final MotionEvent motionEvent) {
+                if (!uiTakeoverEnabled) {
+                    // only log action up or down events as this logging can get quite spammy
+                    if (motionEvent.getAction() == MotionEvent.ACTION_DOWN
+                            || motionEvent.getAction() == MotionEvent.ACTION_UP) {
+                        Log.trace(
+                                ServiceConstants.LOG_TAG,
+                                TAG,
+                                "UI takeover is false, passing the %s event to the parent"
+                                        + " activity.",
+                                MotionEvent.actionToString(motionEvent.getAction()));
+                    }
+                    parentActivity.dispatchTouchEvent(motionEvent);
+                }
+                return super.onTouchEvent(motionEvent);
```

Comment:
- Is `Dialog#onTouchEvent` invoked only when the touch happens outside of the Dialog bounds? Or is it for all touch events on the Dialog?
- It seems like the touch event is being dispatched to the `parentActivity` as well as the dialog if `uiTakeOverEnabled` is false. I believe this will cause a double touch behavior (with one touch event directed to the activity which may or may not be handled and, one to the dialog hosting the message). Is this intentional?  or did you want to do `return parentActivity.dispatchTouchEvent(...)` ?

Reply:
the Dialog#onTouchEvent will only be called when a touch occurs outside of the in-app message webview / dialog bounds as only the webview has a touchListener set for it. [A touch listener was previously set for the decor view](https://github.com/adobe/aepsdk-core-android/pull/587/files#diff-f3361547bd53ca028299d9a6667db93ac72c04dc6164dd23671685e19ecea044L348) (containing the webview and the ui outside of the webview) but this code has been removed.